hawkbit-rest-core/src/main/java/org/eclipse/hawkbit/rest/util/FileStreamingUtil.java

```diff
@@ -0,0 +1,444 @@
+/**
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.rest.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.math.RoundingMode;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.hawkbit.artifact.repository.model.AbstractDbArtifact;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+
+import com.google.common.base.Preconditions;
+import com.google.common.io.ByteStreams;
+import com.google.common.math.DoubleMath;
+
+/**
+ * Utility class for artifact file streaming.
+ */
+public final class FileStreamingUtil {
+
+    private static final Logger LOG = LoggerFactory.getLogger(FileStreamingUtil.class);
+
+    /**
+     * File suffix for MDH hash download (see Linux md5sum).
+     */
+    public static final String ARTIFACT_MD5_DWNL_SUFFIX = ".MD5SUM";
+
+    private static final int BUFFER_SIZE = 0x2000; // 8k
+
+    private FileStreamingUtil() {
+
+    }
+
+    /**
+     * Write a md5 file response.
+     *
+     * @param response
+     *            the response
+     * @param md5Hash
+     *            of the artifact
+     * @param filename
+     *            as provided by the client
+     * @return the response
+     * @throws IOException
+     *             cannot write output stream
+     */
+    public static ResponseEntity<Void> writeMD5FileResponse(final HttpServletResponse response, final String md5Hash,
+            final String filename) throws IOException {
+
+        if (md5Hash == null) {
+            return ResponseEntity.notFound().build();
+        }
+
+        final StringBuilder builder = new StringBuilder();
+        builder.append(md5Hash);
+        builder.append("  ");
+        builder.append(filename);
+        final byte[] content = builder.toString().getBytes(StandardCharsets.US_ASCII);
+
+        final StringBuilder header = new StringBuilder().append("attachment;filename=").append(filename)
+                .append(ARTIFACT_MD5_DWNL_SUFFIX);
+
+        response.setContentLength(content.length);
+        response.setHeader("Content-Disposition", header.toString());
+
+        response.getOutputStream().write(content, 0, content.length);
```

Comment:
You can go for `write(content)` => this is exactly the same and you'll save some characters